frotz.py

```diff
@@ -187,30 +222,54 @@ def _paste_cb(self, button):
         self._vte.paste_clipboard()
 
     def __key_press_cb(self, window, event):
-        if event.state & gtk.gdk.CONTROL_MASK and event.state & gtk.gdk.SHIFT_MASK:
+        if event.state & Gdk.ModifierType.CONTROL_MASK and event.state & Gdk.ModifierType.SHIFT_MASK:
         
-            if gtk.gdk.keyval_name(event.keyval) == "C":
+            if Gdk.keyval_name(event.keyval) == "C":
                 if self._vte.get_has_selection():
                     self._vte.copy_clipboard()              
                 return True
-            elif gtk.gdk.keyval_name(event.keyval) == "V":
+            elif Gdk.keyval_name(event.keyval) == "V":
                 self._vte.paste_clipboard()
                 return True
-                
+
         return False
 
-class VTE(vte.Terminal):
-    def __init__(self):
-        vte.Terminal.__init__(self)
-        self._configure_vte()
 
-        #os.chdir(os.environ["HOME"])
-        self.fork_command()
+class VTE(Vte.Terminal):
 
-    def _configure_vte(self):
-        conf = ConfigParser.ConfigParser()
+    def __init__(self):
+        Vte.Terminal.__init__(self)
+        self.configure_terminal()
+
+        # os.chdir(os.environ["HOME"])
+        if hasattr(Vte.Terminal, "spawn_sync"):
+            self.spawn_sync(
+                Vte.PtyFlags.DEFAULT,
+                os.environ["HOME"],
+                ["/bin/bash"],
+                [],
+                GLib.SpawnFlags.DO_NOT_REAP_CHILD,
+                None,
+                None)
+        else:
+            self.fork_command_full(
```

Comment:
Why is it necessary to support versions of Vte without spawn_sync?

Reply:
Actually I took terminal activity as reference. Terminal uses fork_command_full and spawn_sync https://github.com/sugarlabs/terminal-activity/blob/master/terminal.py#L496.